### DISCUSSION TITLE
navigation history stack

### DISCUSSION THREAD

**Discussion by jet10000** [flet-dev/flet#376] [General]
```
page.go('/settings')
```

Is it possible to set an option to disable the browser's forward and back buttons? Use app's navigation only.

**Comment by FeodorFitsner** [comment]
Well, that was the whole point of Navigation 2.0 implementation in Flutter. Updating URL without pushing a history item was possible in Navigator 1.0. Not sure if we are going to pursue that way soon.

**Reply by jet10000** [reply]
Sharing app URL links, many of them are through social app, and the app url is accessed through the built-in browser of social app. If push to browser history, a navigation bar will be displayed at the bottom, which leads to a very bad experience.

Some social platforms, built-in browsers can't disable this feature via javascript patch unless you don't push to history. Expect to have the option to both update the url and disable push to history.

```python
    def open_login(e):
        page.go("/login")
```

change to

```pyhton
    def open_login2(e):
        print(e)
        page.views.append(auth_view)
        page.update()
```

The above code avoids the push history, but the url cannot be updated.

**Reply by jet10000** [reply]
<img width="293" alt="image" src="https://user-images.githubusercontent.com/2258120/191930002-c63bdfdd-6d3a-4e5b-abed-07bd1ee32c6f.png">

Is it possible to skip pushing to window.history and manually set the url changes?

**Comment by hololeo** [comment]
i would hack that at the javascript level on the html page hosting the flet app

https://stackoverflow.com/questions/12381563/how-can-i-stop-the-browser-back-button-using-javascript

**Reply by jet10000** [reply]
I tried some of these codes yesterday but it didn't work for my use case. Can you share your code.

**Reply by hololeo** [reply]
i havent tried it. why not get it to work without flet involved first to see if its possible and meets your needs
